src/main/webapp/js/communication/geppettoJupyter/PythonControlledCapability.js

```diff
@@ -125,9 +126,37 @@ define(function (require) {
                             }
                             break;
                     }
+                    this.setErrorAlert(this.state.value)
                 }
 
-
+                setErrorAlert(value) {
+                    switch (this.props.realType) {
+                        case 'func':
+                            if (value!="" && value!=undefined) {
+                                Utils.sendPythonMessage("netpyne_geppetto.validateFunction", [value]).then((response) => {
+                                    if (response===false) {
+                                        this.setState({errorMsg: "Not a valid function"})
+                                    }
+                                    else {
+                                        this.setState({errorMsg: ""})
+                                    }
+                                });
+                            }
+                            else {
+                                this.setState({errorMsg: ""})
+                            }
+                            break;
+                        case 'float':
```

Comment:
This doesn't make too much sense if you only add the errorMsg property when realType is equal func, right? See => https://github.com/openworm/org.geppetto.frontend/pull/814/files#diff-97dec52ed194ce91a5d273b6e0c76651R261

Reply:
I need that line in case the user decides to delete the content of the field, leaving **value=""**. So with that line, I reset the error message, and then I assign "1" to the field in order to avoid the   field getting the value ""

Reply:
Not sure about this... You change the errorMsg in the state 'this.setState({errorMsg: ""})' for floats but then in the render, you only use the error message if the realtype is a func.
```                 
if (this.props.realType=='func') {
  wrappedComponentProps['errorText'] = this.state.errorMsg;
}
```
What am I missing?